x-pack/plugins/security_solution/public/resolver/view/panels/copyable_panel_field.tsx

```diff
@@ -48,39 +49,41 @@ export const CopyablePanelField = memo(
     const onMouseEnter = () => setIsOpen(true);
     const onMouseLeave = () => setIsOpen(false);
 
-    const ButtonContent = memo(() => (
-      <StyledCopyableField
-        backgroundColor={copyableFieldBackground}
-        data-test-subj="resolver:panel:copyable-field"
-        activeBackgroundColor={linkColor}
-        onMouseEnter={onMouseEnter}
-      >
-        {content}
-      </StyledCopyableField>
-    ));
+    const hoverArea = useMemo(
+      () => (
+        <StyledCopyableField
+          backgroundColor={copyableFieldBackground}
+          data-test-subj="resolver:panel:copyable-field-hover-area"
+          activeBackgroundColor={linkColor}
+          onMouseEnter={onMouseEnter}
+        >
+          {content}
+        </StyledCopyableField>
+      ),
+      [content, copyableFieldBackground, linkColor]
+    );
 
-    const onClick = useCallback(
-      async (event: React.MouseEvent<HTMLButtonElement>) => {
-        try {
-          await navigator.clipboard.writeText(textToCopy);
-        } catch (error) {
-          if (toasts) {
-            toasts.addError(error, {
-              title: i18n.translate('xpack.securitySolution.resolver.panel.copyFailureTitle', {
-                defaultMessage: 'Copy Failure',
-              }),
-            });
-          }
+    const { writeTextToClipboard } = useContext(SideEffectContext);
+
+    const onClick = useCallback(async () => {
+      try {
+        await writeTextToClipboard(textToCopy);
+      } catch (error) {
+        if (toasts) {
+          toasts.addError(error, {
+            title: i18n.translate('xpack.securitySolution.resolver.panel.copyFailureTitle', {
+              defaultMessage: 'Copy Failure',
+            }),
+          });
         }
-      },
-      [textToCopy, toasts]
-    );
+      }
+    }, [textToCopy, toasts, writeTextToClipboard]);
 
     return (
       <div onMouseLeave={onMouseLeave}>
         <EuiPopover
           anchorPosition={'downCenter'}
-          button={<ButtonContent />}
+          button={hoverArea}
```

Comment:
Not sure if this is something you want to resolve in this PR, but all this hover interaction (and whatever is happening on `onMouseLeave` above) can't really be made accessibly. 

I'm not super familiar with the security app so would need some help getting started to offer possible solutions but feel free to reach out with any questions.

Reply:
This gif shows the feature in use. On hover, gray backgrounds are added to `div`s. On `mouseenter`, the a tooltip shows up near the div. The tooltip contains a button with an image in it. You might not know it is a button since it has no background, border, or label. If the user hovers with the mouse over the button, a tooltip shows up on top of the original tooltip. This tooltip describes the button. Clicking this will attempt to copy text to the users clipboard.

![copy-to-clipboard-feature mov](https://user-images.githubusercontent.com/35559/96737013-34136400-138b-11eb-982c-99a195c1d28e.gif)

Reply:
Actually, I think there's a pretty straight forward solution to all this! 🤞 

If `StyledCopyableField` becomes a `button` instead of a `div`, you can trigger the same popover on click (which works for keyboard "clicks" as well) and leave the same hover trigger you have now. This will also give you some basic focus states for free though you may want to style your own if you don't like the defaults.

You can also mimic the `mouseenter` effect on the whole panel by adding a `StyledPanel:focus-within` rule in addition to the `:hover` rule which will trigger when any children of the panel are focused.

Lastly, we can improve the accessibility of the popover by adding the `ownFocus` (or `ownFocus={true}`) attribute. (Honestly, this prop can/should be applied to every `EuiPopover` but isn't right now due to some historical reasons in EUI...)

Reply:
Stepping out of strictly accessibility for a sec, there seems to be enough room in the screenshot you posted to add the copy button inline to the right of the timestamps. That would remove a lot of the a11y concerns we're dealing with now if it's a viable option...

Reply:
We discussed doing that, but were looking to maintain parity with our current consuming application (security_solution) in how they have copy/paste functionality implemented. I know there's work planned to make all of this generally more accessible. Thoughts @monina-n @lindseypoli?

Reply:
🔴 I already put my thoughts in on this on a previous PR, but just to reiterate , I think the problems with putting something in that keyboard users can't use overrides the style/ux consistency concerns.